Makefile

```diff
@@ -0,0 +1,76 @@
+VERSION := 0.19.1
+
+# install directory layout
+PREFIX ?= /usr/local
+INCLUDEDIR ?= $(PREFIX)/include
+LIBDIR ?= $(PREFIX)/lib
+PCLIBDIR ?= $(LIBDIR)/pkgconfig
+
+# collect C++ sources, and link if necessary
+CPPSRC := $(wildcard src/*.cc)
+
+ifeq (, $(CPPSRC))
+	ADDITIONALLIBS := 
+else
+	ADDITIONALLIBS := -lc++
+endif
+
+# collect sources
+SRC := $(wildcard src/*.c)
+SRC += $(CPPSRC)
+OBJ := $(addsuffix .o,$(basename $(SRC)))
+
+# ABI versioning
+SONAME_MAJOR := 0
+SONAME_MINOR := 0
+
+CFLAGS ?= -O3 -Wall -Wextra -Werror
+CXXFLAGS ?= -O3 -Wall -Wextra -Werror
+override CFLAGS += -std=gnu99 -fPIC
+override CXXFLAGS += -fPIC
+
+# OS-specific bits
+ifeq ($(shell uname),Darwin)
+	SOEXT = dylib
+	SOEXTVER_MAJOR = $(SONAME_MAJOR).dylib
+	SOEXTVER = $(SONAME_MAJOR).$(SONAME_MINOR).dylib
+	LINKSHARED += -dynamiclib -Wl,$(ADDITIONALLIBS),-install_name,$(LIBDIR)/libtree-sitter-go.$(SONAME_MAJOR).dylib
+else
+	SOEXT = so
+	SOEXTVER_MAJOR = so.$(SONAME_MAJOR)
+	SOEXTVER = so.$(SONAME_MAJOR).$(SONAME_MINOR)
+	LINKSHARED += -shared -Wl,$(ADDITIONALLIBS),-soname,libtree-sitter-go.so.$(SONAME_MAJOR)
```

Comment:
When `$(ADDITIONALLIBS)` is  empty, it will get an error for some versions of `ld`.

![image](https://user-images.githubusercontent.com/15541038/147742002-a60a76aa-32b3-49f6-9aba-d5549ef4c94a.png)

Such as version: GNU ld (GNU Binutils for Debian) 2.25.

Reply:
As there are no C++ resources, can we remove this?

Reply:
I'm sorry about this. It was really challenging to produce a Makefile that worked consistently for both C/C++ parsers. I'm not sure if the maintainers would prefer. I'll check, and get back to this as a separate issue.

Reply:
Thanks！